elytron/src/main/java/org/wildfly/extension/elytron/ModifiableRealmDecorator.java

```diff
@@ -543,41 +546,48 @@ protected void executeRuntimeStep(final OperationContext context, final ModelNod
         }
 
         private Password createPassword(final OperationContext parentContext, final String principalName, String passwordType, ModelNode passwordNode) throws OperationFailedException, NoSuchAlgorithmException, InvalidKeySpecException  {
+
+            ModifiableSecurityRealm fsRealm = getModifiableSecurityRealm(parentContext);
+            Charset hashCharset = StandardCharsets.UTF_8;
+            if (fsRealm instanceof FileSystemSecurityRealm) {
```

Comment:
We also need to think about the LDAP realm case here.

Reply:
The LdapSecurityRealm is not a public class, so I am not sure how to access it's charset. I thought about adding the getter to ModifiableSecurityRealm, but I think that's a public API so I am not sure whether I can edit that. Should I add the method to ModifiableSecurityRealm with a default implementation?

Reply:
I'll ping you to discuss this one a bit more.